gameplay/src/Quaternion.cpp

```diff
@@ -80,6 +101,17 @@ void Quaternion::createFromAxisAngle(const Vector3& axis, float angle, Quaternio
     dst->w = cosf(halfAngle);
 }
 
+void Quaternion::computeEuler(float* yaw, float* pitch, float* roll)
+{
+	GP_ASSERT(yaw);
+	GP_ASSERT(pitch);
+	GP_ASSERT(roll);
+
+	*pitch = std::atan2(2 * (w*x + y*z), 1 - 2 * (x*x + y*y));
```

Comment:
Unfortunately the Euler angle representation of a 3D rotation has an inherent singularity, and you can see that here for yaw angles of +/- 90 degrees. Consider a valid unit quaternion with w=1/sqrt(2), x=0, y=1/sqrt(2), z=0. Your atan2() calls will both return NaN due to an attempt to call atan2(0, 0). The question is how do we deal with this inherent singularity. One approach might be to detect the singularity and return pre-determined values for pitch and roll which might be either hard-coded or provided by the caller via additional parameters to this function. Thoughts?